File: mac/src/Beecell/Models/BeecellState.swift
import Foundation

public struct BeecellState: Codable, Equatable {
    public var freeCells: [Pile]        // 4 (1-deck) or 8 (2-decks) cells
    public var foundations: [Pile]     // 4 (1-deck) or 8 (2-decks) foundations
    public var tableau: [Pile]         // 8 (1-deck) or 10 (2-decks) columns
    public var score: Int
    public var movesCount: Int
    public var timerSeconds: Int
    public var isTimerActive: Bool
    public var hasWon: Bool
    
    public init(
        freeCells: [Pile],
        foundations: [Pile],
        tableau: [Pile],
        score: Int,
        movesCount: Int,
        timerSeconds: Int,
        isTimerActive: Bool,
        hasWon: Bool
    ) {
        self.freeCells = freeCells
        self.foundations = foundations
        self.tableau = tableau
        self.score = score
        self.movesCount = movesCount
        self.timerSeconds = timerSeconds
        self.isTimerActive = isTimerActive
        self.hasWon = hasWon
    }
}

File: android/app/src/main/java/com/leah/honeycomb/beecell/BeecellState.kt
package com.leah.honeycomb.beecell

import com.leah.honeycomb.Pile
import kotlinx.serialization.Serializable

@Serializable
data class BeecellState(
    val freeCells: List<Pile> = emptyList(),
    val foundations: List<Pile> = emptyList(),
    val tableau: List<Pile> = emptyList(),
    val movesCount: Int = 0,
    val hasWon: Boolean = false,
    val timerSeconds: Int = 0,
    val isTimerActive: Boolean = false
)
